bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreClientImpl.java

```diff
@@ -0,0 +1,594 @@
+      handleBigQueryRetryException(e);
+    }
+
+    return response;
+  }
+
+  @SuppressWarnings("FormatStringAnnotation")
+  private Table internalCreate(Table table) {
+    try {
+      HttpResponse response =
+          client
+              .tables()
+              .insert(
+                  Preconditions.checkNotNull(table.getTableReference()).getProjectId(),
+                  Preconditions.checkNotNull(table.getTableReference()).getDatasetId(),
+                  table)
+              .executeUnparsed();
+      return convertExceptionIfUnsuccessful(response).parseAs(Table.class);
+    } catch (IOException e) {
+      throw new RuntimeIOException(e);
+    } catch (AlreadyExistsException e) {
+      throw new AlreadyExistsException(String.format("Table %s already exists", table), e);
+    }
+  }
+
+  @Override
+  @SuppressWarnings("FormatStringAnnotation")
+  public Table load(TableReference tableReference) {
+    try {
+      HttpResponse response =
+          client
+              .tables()
+              .get(
+                  tableReference.getProjectId(),
+                  tableReference.getDatasetId(),
+                  tableReference.getTableId())
+              .executeUnparsed();
+      if (response.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+        throw new NoSuchTableException(response.getStatusMessage());
+      }
+
+      return validateTable(convertExceptionIfUnsuccessful(response).parseAs(Table.class));
+    } catch (IOException e) {
+      throw new RuntimeIOException(e);
+    }
+  }
+
+  @Override
+  @SuppressWarnings("FormatStringAnnotation")
+  public Table update(TableReference tableReference, Table table) {
+    // Ensure it is an Iceberg table supported by the BQ metastore catalog.
+    validateTable(table);
+
+    ExternalCatalogTableOptions newExternalCatalogTableOptions =
+        new ExternalCatalogTableOptions()
+            .setStorageDescriptor(table.getExternalCatalogTableOptions().getStorageDescriptor())
+            .setConnectionId(table.getExternalCatalogTableOptions().getConnectionId())
+            .setParameters(table.getExternalCatalogTableOptions().getParameters());
+    Table updatedTable =
+        new Table()
+            .setExternalCatalogTableOptions(newExternalCatalogTableOptions)
+            // Must set the schema as null for using schema auto-detect.
+            .setSchema(Data.nullOf(TableSchema.class));
+
+    Table response = null;
+    try {
+      response =
+          BigQueryRetryHelper.runWithRetries(
+              () -> internalUpdate(tableReference, updatedTable, table.getEtag()),
+              bigqueryOptions.getRetrySettings(),
+              BIGQUERY_EXCEPTION_HANDLER,
+              bigqueryOptions.getClock(),
+              DEFAULT_RETRY_CONFIG);
+    } catch (BigQueryRetryHelper.BigQueryRetryHelperException e) {
+      handleBigQueryRetryException(e);
+    }
+
+    return response;
+  }
+
+  @SuppressWarnings("FormatStringAnnotation")
```

Comment:
same as above. please also update alll the other places in this class

Reply:
I removed some of them But I can not remove all because in some function i just re-throw exception with wrapping cause.

Reply:
All of them can be removed and the exception instantiation can be updated to `throw new NoSuchTableException("%s", response.getStatusMessage())`. We also use the same approach in other places in the codebase, such as in `ErrorHandlers`